app/src/main/java/ch/epfl/sdp/game/Game.java

```diff
@@ -126,7 +126,7 @@ public void addToDisplayList(Displayable displayable) {
      * @param displayable The displayable to be removed.
      */
     public void removeFromDisplayList(Displayable displayable) {
-        displayable.unDisplayOn(mapApi);
+        renderer.unDisplay(displayable);
```

Comment:
What is not good with the .unDisplay(mapApi)?

Reply:
We need to go through the renderer as it's the MapsActivity and it can call runOnUiThread (.unDisplayOn was not run on the main thread)